### PR TITLE
Remove remaining ICodeLoader Interfaces

### PR DESCRIPTION
**Pull request by RishhiB**
The deprecated ICodeLoader Interface should be removed.

## Approach
The interface `ICodeLoader` was deprecated a while ago in previous releases. The alternative for `ICodeLoader` interface is the `ICodeDetailsLoader` interface which can be imported from `@fluidframework/container-definitions`.

## Steps
The alternative for `ICodeLoader` interface is the `ICodeDetailsLoader` interface which can be imported from `@fluidframework/container-definitions`.

## Does this introduce a breaking change?

-   [X] Yes
-   [ ] No

## Other information or known dependencies

[AB#346](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/346)
PR: #9657, #8193

### CODE REVIEW

**Comment by sonalivdeshpande** [comment]
maybe try adding `"InterfaceDeclaration_ICodeLoader": {"forwardCompat": false}` to container-definition package.json and doing a local build again

File: packages/common/container-definitions/src/test/types/validateContainerDefinitionsPrevious.ts [hunk 1]
```diff
@@ -325,30 +325,6 @@ declare function use_old_InterfaceDeclaration_ICodeDetailsLoader(
 use_old_InterfaceDeclaration_ICodeDetailsLoader(
     get_current_InterfaceDeclaration_ICodeDetailsLoader());
 
-/*
-* Validate forward compat by using old type in place of current type
-* If breaking change required, add in package.json under typeValidation.broken:
-* "InterfaceDeclaration_ICodeLoader": {"forwardCompat": false}
-*/
-declare function get_old_InterfaceDeclaration_ICodeLoader():
```
Comment (sonalivdeshpande) [inline packages/common/container-definitions/src/test/types/validateContainerDefinitionsPrevious.ts:333]: was this file change manually? can you try reverting this file changes and do a local build again?

test/types/** files are auto generated on build

**Review by sonalivdeshpande — CHANGES_REQUESTED** [review]

**Review by sonalivdeshpande — APPROVED** [review]

File: BREAKING.md [hunk 2]
```diff
@@ -88,6 +88,7 @@ It has been removed in this release--consumers should assume any raised delta ev
 - [Remove `noopTimeFrequency` and `noopCountFrequency` from ILoaderOptions](#remove-nooptimefrequency-and-noopcountfrequency-from-iloaderoptions)
 - [proxyLoaderFactories members removed from ILoaderProps and ILoaderServices](#proxyloaderfactories-members-to-be-removed-from-iloaderprops-and-iloaderservices)
 - [IContainer.connectionState yields finer-grained ConnectionState values](#icontainerconnectionstate-yields-finer-grained-connectionstate-values)
+- [Remove ICodeLoader from container-definitions](#Remove-ICodeLoader-from-container-definitions)
```
Comment (sonalivdeshpande) [inline BREAKING.md:91]: nits: [Remove ICodeLoader from `@fluidframework/container-definitions`]

File: BREAKING.md [hunk 3]
```diff
@@ -189,6 +190,10 @@ Additionally, please note that the `Connecting` state is being renamed to `Catch
 `ConnectionState.Connecting` is marked as deprecated, please use `ConnectionState.CatchingUp` instead.
 `ConnectionState.Connecting` will be removed in the following major release.
 
+### Remove ICodeLoader from container-definitions
+`ICodeLoader` in `@fluidframework/container-definitions` was deprecated a while ago and is now removed. Use `ICodeDetailsLoader` by importing it from `@fluidframework/container-loader`.
```
Comment (scottn12) [inline BREAKING.md:194]: Nits:

> `ICodeLoader` in `@fluidframework/container-definitions` was deprecated a while ago and is now removed.

Can we replace "a while ago" with the exact version? Either that or just remove it all together

```suggestion
`ICodeLoader` in `@fluidframework/container-definitions` was deprecated a while ago and is now removed. Use `ICodeDetailsLoader` from `@fluidframework/container-loader` instead.
```

File: BREAKING.md [hunk 4]
```diff
@@ -88,6 +88,7 @@ It has been removed in this release--consumers should assume any raised delta ev
 - [Remove `noopTimeFrequency` and `noopCountFrequency` from ILoaderOptions](#remove-nooptimefrequency-and-noopcountfrequency-from-iloaderoptions)
 - [proxyLoaderFactories members removed from ILoaderProps and ILoaderServices](#proxyloaderfactories-members-to-be-removed-from-iloaderprops-and-iloaderservices)
 - [IContainer.connectionState yields finer-grained ConnectionState values](#icontainerconnectionstate-yields-finer-grained-connectionstate-values)
+- [Remove ICodeLoader from @fluidframework/container-definitions](#Remove-ICodeLoader-from-container-definitions)
```
Comment (sonalivdeshpande) [inline BREAKING.md:91]: this is part of 1.0.0 breaking changes. this should this be included under 2.0.0 breaking change section :) 

link needs to be fixed here: 
- [Remove ICodeLoader from `@fluidframework/container-definitions`](#Remove-ICodeLoader-from-fluidframeworkcontainer-definitions)

File: BREAKING.md [hunk 5]
```diff
@@ -189,6 +190,10 @@ Additionally, please note that the `Connecting` state is being renamed to `Catch
 `ConnectionState.Connecting` is marked as deprecated, please use `ConnectionState.CatchingUp` instead.
 `ConnectionState.Connecting` will be removed in the following major release.
 
+### Remove ICodeLoader from container-definitions
```
Comment (sonalivdeshpande) [inline BREAKING.md:193]: here as well -- - Remove ICodeLoader from `@fluidframework/container-definitions`

File: BREAKING.md [hunk 6]
```diff
@@ -189,6 +190,10 @@ Additionally, please note that the `Connecting` state is being renamed to `Catch
...
+`ICodeLoader` in `@fluidframework/container-definitions` was deprecated a while ago and is now removed. Use `ICodeDetailsLoader` by importing it from `@fluidframework/container-loader`.
+Please refer to [replace ICodeLoader with ICodeDetailsLoader interface](#Replace-ICodeLoader-with-ICodeDetailsLoader-interface) for more details.
```
Comment (scottn12) [inline BREAKING.md:195]: I think it's probably better to just copy the details from the section you referenced so people don't have to go searching for it

**Review by scottn12 — APPROVED** [review]
Looks good! Just a couple nits below and we should be good to go!

**Comment by sonalivdeshpande** [comment]
Let me know if you need help merging this PR in 😃

**Comment by sonalivdeshpande** [comment]
@RishhiB -- merging this pr in

**Pull request merged**